File: backend/src/main/java/com/fooddelivery/service/AdminService.java
package com.fooddelivery.service;

import com.fooddelivery.dto.DashboardStatsResponse;
import com.fooddelivery.dto.OrderAnalyticsResponse;
import com.fooddelivery.dto.OrderResponse;
import com.fooddelivery.dto.RevenueAnalyticsResponse;
import com.fooddelivery.dto.UpdateUserStatusRequest;
import com.fooddelivery.dto.UserResponse;
import com.fooddelivery.entity.OrderStatus;
import com.fooddelivery.entity.Role;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Service interface for admin dashboard, user management,
 * restaurant management, and analytics.
 */
public interface AdminService {

    // ---- Dashboard ----

    DashboardStatsResponse getDashboardStats();

    // ---- User Management ----

    Page<UserResponse> getAllUsers(Pageable pageable);

    Page<UserResponse> searchUsers(String search, Pageable pageable);

    Page<UserResponse> getUsersByRole(Role role, Pageable pageable);

    UserResponse getUserById(Long id);

    UserResponse updateUserStatus(Long id, UpdateUserStatusRequest request);

    void deleteUser(Long id);

    // ---- Restaurant Management ----

    Page<com.fooddelivery.dto.RestaurantResponse> getAllRestaurants(Pageable pageable);

    void approveRestaurant(Long restaurantId);

    void disableRestaurant(Long restaurantId);

    void deleteRestaurant(Long restaurantId);

    // ---- Order Management ----

    Page<OrderResponse> getAllOrders(Pageable pageable);

    Page<OrderResponse> getOrdersByStatus(OrderStatus status, Pageable pageable);

    Page<OrderResponse> getOrdersByRestaurant(Long restaurantId, Pageable pageable);

    OrderResponse getOrderById(Long orderId);

    // ---- Analytics ----

    OrderAnalyticsResponse getOrderAnalytics();

    List<RevenueAnalyticsResponse> getDailyRevenue(int days);
}

File: frontend/src/services/adminService.js
/**
 * Admin API service — maps to AdminController.
 */
import api from '../api/axios';

const URL = '/api/admin';

const adminService = {
  // ---- Dashboard ----
  getDashboardStats: () => api.get(`${URL}/dashboard`).then((r) => r.data),

  // ---- User Management ----
  getUsers: (params = {}) =>
    api.get(`${URL}/users`, { params }).then((r) => r.data),

  getUserById: (id) => api.get(`${URL}/users/${id}`).then((r) => r.data),

  updateUserStatus: (id, data) =>
    api.put(`${URL}/users/${id}/status`, data).then((r) => r.data),

  deleteUser: (id) => api.delete(`${URL}/users/${id}`),

  // ---- Restaurant Management ----
  getRestaurants: (page = 0, size = 10) =>
    api.get(`${URL}/restaurants`, { params: { page, size } }).then((r) => r.data),

  approveRestaurant: (id) =>
    api.put(`${URL}/restaurants/${id}/approve`).then((r) => r.data),

  disableRestaurant: (id) =>
    api.put(`${URL}/restaurants/${id}/disable`).then((r) => r.data),

  deleteRestaurant: (id) => api.delete(`${URL}/restaurants/${id}`),

  // ---- Order Management ----
  getOrders: (params = {}) =>
    api.get(`${URL}/orders`, { params }).then((r) => r.data),

  getOrderById: (id) => api.get(`${URL}/orders/${id}`).then((r) => r.data),

  // ---- Analytics ----
  getOrderAnalytics: () =>
    api.get(`${URL}/analytics/orders`).then((r) => r.data),

  getRevenueAnalytics: (days = 30) =>
    api.get(`${URL}/analytics/revenue`, { params: { days } }).then((r) => r.data),
};

export default adminService;
